dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/deduplication/DefaultDeduplicationService.java

```diff
@@ -127,17 +130,24 @@ public void manualMerge(DeduplicationMergeParams deduplicationMergeParams)
     merge(deduplicationMergeParams);
   }
 
-  private String getAutoMergeConflictErrors(TrackedEntity original, TrackedEntity duplicate) {
+  private void validateCanBeAutoMerged(DeduplicationMergeParams params)
+      throws PotentialDuplicateConflictException {
+    TrackedEntity original = params.getOriginal();
+    TrackedEntity duplicate = params.getDuplicate();
+
     if (!original.getTrackedEntityType().equals(duplicate.getTrackedEntityType())) {
-      return "Entities have different Tracked Entity Types.";
+      throw new PotentialDuplicateConflictException(
+          "PotentialDuplicate cannot be merged automatically: Entities have different Tracked Entity Types.");
```

Comment:
can we have the first part of this sentence stored in a String so we don't repeat it 4 times?

Reply:
we can